flang/include/flang/Optimizer/OpenMP/Passes.td

```diff
@@ -81,6 +81,13 @@ def DoConcurrentConversionPass : Pass<"omp-do-concurrent-conversion", "mlir::fun
   ];
 }
 
+def LowerNontemporalPass : Pass<"lower-nontemporal", "mlir::func::FuncOp"> {
```

Comment:
Why not run this on `omp::SimdOp`? Running it on functions is wasteful because many functions will not contain a simd op and runs the risk of missing any simd ops which are not nested inside of functions (I can't see how that would happen currently but we have had problems before lowering non-function (but function-like) operations such as `omp.declare_reduction` because subsequent passes incorrectly assumed all operations were nested inside of functions.

Reply:
It was only a hypothetical example - I can't think of a case where we would do simd not inside of a function either.

Still, changing this to
```
def LowerNontemporalPass : Pass<"lower-nontemporal", "mlir::omp::SimdOp"> {
```
should be a better fit with the intention of the pass, because it will not modify anything outside of the simd op

Reply:
Re: declaration operations: It could be but I thought that would be so unlikely that it would be better to only make that (minimal) change if it was ever needed.

Re: this thread: Kaviya and I were unable to get it working. It turns out that the MLIR pass manager doesn't allow for arbitrary nesting: only 1 level deep. So you can run a func.func pass on a pass manager for a module because the func.func is a direct descendant. But it couldn't be done for simd because that could be anywhere in the operation tree. Sorry, I should have posted something here.